modin/core/dataframe/base/dataframe/utils.py

```diff
@@ -48,3 +53,71 @@ class JoinType(Enum):  # noqa: PR01
     LEFT = "left"
     RIGHT = "right"
     OUTER = "outer"
+
+
+def join_columns(
+    left: pandas.Index,
+    right: pandas.Index,
+    left_on: Union[Iterable[IndexLabel], IndexLabel],
+    right_on: Union[Iterable[IndexLabel], IndexLabel],
+    suffixes: Tuple[str, str],
+) -> Tuple[pandas.Index, Dict[IndexLabel, IndexLabel], Dict[IndexLabel, IndexLabel]]:
+    """
+    Compute resulting columns for the two dataframes being merged.
+
+    Parameters
+    ----------
+    left : pandas.Index
+        Columns of the left frame to join.
+    right : pandas.Index
+        Columns of the right frame to join.
+    left_on : list-like or scalar
+        Column names on which the frames are joined in the left DataFrame.
+    right_on : list-like or scalar
+        Column names on which the frames are joined in the right DataFrame.
+    suffixes : tuple[str, str]
+        A 2-length sequence containing suffixes to append to the intersected columns.
+
+    Returns
+    -------
+    pandas.Index, dict[IndexLabel -> IndexLabel], dict[IndexLabel -> IndexLabel]
+        Returns columns for the resulting frame and mappings of old to new column
+        names for `left` and `right` accordingly.
+    """
+    left_on, right_on = map(
+        lambda x: cast(Iterable[IndexLabel], [x] if is_scalar(x) else x),
```

Comment:
using `cast` to make `mypy` acknowledged that the variable now ensured to be `Iterable[IndexLabel]`